app/queries/get_linkables.rb

```diff
@@ -33,9 +22,18 @@ def call
           state_fallback_order: [:published, :draft]
         )
 
-        Edition.with_document.includes(:document).where(id: edition_ids).map do |edition|
-          LinkablePresenter.from_hash(edition.to_h.deep_symbolize_keys)
-        end
+        Edition.with_document
+          .where(id: edition_ids)
+          .pluck(:title, :content_id, :state, :base_path, "COALESCE(details->>'internal_name', title)")
+          .map do |(title, content_id, state, base_path, internal_name)|
+            LinkablePresenter.new(
+              title,
+              content_id,
+              state,
+              base_path,
+              internal_name,
```

Comment:
Good point, that would be nicer. I won't bother raising a fresh PR for that now since I'm assuming your work will replace it.